### PR TITLE
Created a testing setup for integration testing

### PR DESCRIPTION
**Pull request by kburger**
Using a spock/geb/phantomjs stack we can now automatically build the docker image and run the tests against it. There are a lot of improvements to be made, but this is my proposal for a starting point.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,8 @@ FROM openjdk:8-alpine
 
 # build and cleanup
 COPY setup.sh /opt/setup.sh
-RUN chmod +x /opt/setup.sh && /opt/setup.sh
```
Comment (Shamanou) [inline Dockerfile:5]: Why not this but instead the two line solution?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+sudo: required
+language: java
+jdk:
+  - oraclejdk8
+services:
+  - docker
+before_install:
+  - sudo apt-get -qq update
```
Comment (Shamanou) [inline .travis.yml:8]: possible more readable alternative for apt-get -qq is apt-get -q=2 (quiet level 2)

**Review by Shamanou — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (kburger) [reply Dockerfile:5]: Running the 2 commands chained together gave some issues on linux, let me see if I can reproduce that.

File: .travis.yml (hunk 2 above)
Comment (kburger) [reply .travis.yml:8]: `-qq` is quiet level 2, with some caveats that do not apply to an `apt-get update` (only `install` iirc). Define more readable?

Comment (Shamanou) [reply .travis.yml:8]: Ah ok i didnt know that,  i just thought that -qq might look less clear then -q=2 but its redundant if -q=2 does not work

File: Dockerfile (hunk 1 above)
Comment (kburger) [reply Dockerfile:5]: Running the chained command on linux:
```
$ uname -vr
4.4.0-64-generic #85-Ubuntu SMP Mon Feb 20 11:50:30 UTC 2017
$ docker -v
Docker version 17.03.1-ce, build c6d412
$ docker build -t test .
Sending build context to Docker daemon 105.5 kB
Step 1/6 : FROM openjdk:8-alpine
 ---> 478bf389b75b
Step 2/6 : COPY setup.sh /opt/setup.sh
 ---> b8cd5c1fb84f
Removing intermediate container ac53165aa10f
Step 3/6 : RUN chmod +x /opt/setup.sh && /opt/setup.sh
 ---> Running in 888f76751718
/bin/sh: /opt/setup.sh: Text file busy
The command '/bin/sh -c chmod +x /opt/setup.sh && /opt/setup.sh' returned a non-zero code: 2
```

Running it on OSX works as intended both ways, so splitting it into two lines is a more cross platform solution. And fortunately it doesn't add (much, as in less than a KB) to the final image size.

Comment (Shamanou) [reply Dockerfile:5]: true,

ok then its a good sollution

**Comment by Shamanou** [comment]
I think the changes are good

**Review by Shamanou — APPROVED** [review]

**Pull request merged**